extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -706,6 +700,13 @@ public void onFetchPage(@Nonnull final Downloader downloader)
                 .done())
                 .getBytes(UTF_8);
 
+        CompletableFuture iosTask = CompletableFuture.supplyAsync(() -> {
```

Comment:
I am not sure this is a good solution: what if the task hasn't finished when `getHlsUrl()` is called? If the extractor was all built with async means it would integrate good, but in this case there is nothing waiting for the async task to complete. Also, futures are not used anywhere else in the extractor, by doing a fast search.